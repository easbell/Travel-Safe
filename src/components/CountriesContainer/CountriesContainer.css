.container {
  height: 800px;
  overflow: scroll;
  margin-top: 48px;
  display: grid;
  grid-template-rows: 150px;
  grid-gap: 30px;
  padding: 0px 35px;
  width: 75%;
  margin: 48px auto;
  grid-template-columns: repeat( auto-fit, minmax(350px, 1fr) );
}

.react-fuzzy-filter__input {
  width: 25%;
  border: 0;
  box-shadow: 2px 2px 11px #818181;
  border-radius: 5px;
  height: 49px;
  padding-left: 15px;
  font-size: 20px;
}