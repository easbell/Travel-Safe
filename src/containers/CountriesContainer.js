import React, { Component } from 'react';
import { connect } from 'react-redux';
import { combineData } from '../helpers/combineData';

export class CountriesContainer extends Component {
  cleanData = () => {
    const { safety, vaccines } = this.props;
    const cleaned = combineData(safety, vaccines);
    console.log(cleaned)
  }

  render() {
    return (
      <div>
        {this.cleanData()}
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  safety: state.safety,
  vaccines: state.vaccines
});

export default connect(mapStateToProps)(CountriesContainer);