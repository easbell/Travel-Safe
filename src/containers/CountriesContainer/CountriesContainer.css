.container {
  height: 800px;
  overflow: scroll;
}