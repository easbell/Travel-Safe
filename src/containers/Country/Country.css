.country {
  display: flex;
  align-items: center;
  justify-content: space-between;
  box-shadow: -4px 4px 11px -5px rgba(97,97,97,1);
  width: 50%;
  margin: 40px auto;
  height: 110px;
  border-radius: 5px;
  cursor: pointer;
  padding: 0px 110px;
  background: #fff;
  opacity: .8;
  color: #000;
}

.country:hover {
  box-shadow: -7px 7px 12px -5px rgba(97,97,97,1);
  opacity: 1;
}

.rating {
  border-radius: 50%;
  height: 55px;
  width: 55px;
  font-size: 22px;
  color: #fff;
  border: 0;
  text-decoration: none;
}

.L {
  background-color: #37d57b;
}

.M {
  background-color: #82b8f7;
}

.H {
  background-color: #fbf282;
}

.E {
  background-color: #ee3649;
}