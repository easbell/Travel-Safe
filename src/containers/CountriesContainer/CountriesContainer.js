import React, { Component } from 'react';
import './CountriesContainer.css';
import Country from '../Country/Country';
import fuzzyFilterFactory from "react-fuzzy-filter";
const { InputFilter, FilterResults } = fuzzyFilterFactory();

export class CountriesContainer extends Component {
  renderCountries = (filtered) => {
    return filtered.map(country => {
      return <Country key={country.name} {...country}/>
    });
  }

  render() {
    const { countries } = this.props;
    const fuseConfig = {
      keys: ['name'],
      matchAllTokens: true,
      tokenize: true,
      threshold: 0
    };

    return (
      <div>
        {countries.length === 57 && <InputFilter />}
        <FilterResults items={countries} fuseConfig={fuseConfig}>
          {filteredItems => {
            return (
              <div className='container'>
                {this.renderCountries(filteredItems)}
              </div>
            )
          }}
        </FilterResults>
      </div>
    )
  }
}

export default CountriesContainer;