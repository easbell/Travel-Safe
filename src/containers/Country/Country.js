import React, { Component } from 'react';
import classnames from 'classnames';
import './Country.css';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import cogoToast from 'cogo-toast';

export class Country extends Component {
  constructor() {
    super();
    this.state = {
      rating: '',
      saved: false
    }
  }

  componentDidMount() {
    this.checkRating();
    this.checkSaved();
  }

  checkSaved = ()  => {
    const { id } = this.props
    const localSaved = localStorage.getItem('saved')
    if(localSaved && localSaved.includes(id)) {
      this.setState({ saved: true })
    }
  }

  checkRating = () => {
    const { rating } = this.props;
    if(rating <= 2.5) {
      this.setState({rating: 'L'})
    } else if(2.6 <= rating && rating <= 3.5) {
      this.setState({rating: 'M'})
    } else if(3.6 <= rating && rating <= 4.5) {
      this.setState({rating: 'H'})
    } else if(rating > 4.5) {
      this.setState({rating: 'E'})
    }
  }

  displayMessage = (message) => {
    cogoToast.success(message, { 
      position: 'bottom-left'
    });
  }

  addCountry = () => {
    const { id } = this.props;
    const localSaved = JSON.parse(localStorage.getItem('saved'))
    if(localSaved && !localSaved.includes(id)) {
      localStorage.setItem('saved', JSON.stringify([...localSaved, id]));
    } else if(!localSaved) {
      localStorage.setItem('saved', JSON.stringify([id]));
    }
    this.displayMessage('Country was added to favorites.')      
  }

  toggleCountry = (e) => {
    const { id } = this.props;
    if(e.target.className.includes('countrySaved')) {
      this.setState({ saved: false });
      const localSaved = JSON.parse(localStorage.getItem('saved'))
      const filteredSaved = localSaved.filter(country => country !== id)
      localStorage.setItem('saved', JSON.stringify(filteredSaved));
      this.displayMessage('Country removed from favorites.')
    } else {
      this.setState({ saved: true });
      this.addCountry();
    }
  }
  
  render() {
    const classNames = require('classnames');
    const { id, name } = this.props;
    const { rating, saved } = this.state;
    const btnClass = classNames({
      btn: true,
      add: true,
      countrySaved: saved
    });

    return (
      <div className='country'>
        {saved && <button className={btnClass} onClick={this.toggleCountry}>-</button>}
        {!saved && <button className={btnClass} onClick={this.toggleCountry}>+</button>}
        <Link to={`/details/${id}`} style={{ textDecoration: 'none' }}>
          <h2>{name}</h2>
        </Link>
        <button className={classnames('btn', rating)}>{rating}</button>
      </div>
    )
  }
}

export const mapStateToProps = state => ({
  saved: state.saved
})

export default connect(mapStateToProps)(Country);