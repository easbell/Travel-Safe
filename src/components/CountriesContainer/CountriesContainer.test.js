import React from 'react';
import { CountriesContainer } from './CountriesContainer';
import { shallow } from 'enzyme';
import { Country } from '../../containers/Country/Country';

describe('CountriesContainer', () => {
  let wrapper;
  let mockCountries;
  beforeEach(() => {
    mockCountries = [{name: 'USA'}, {name: 'France'}];
    wrapper = shallow(
      <CountriesContainer countries={mockCountries}/>)
  });

  it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
  });

  it.skip('should call renderCountries', () => {
    const mockCountries = [{name: 'USA'}, {name: 'France'}];
    const expected = [<Country id="USA" name="USA" />, <Country id="France" name="France" />]
    const cleaned = wrapper.instance().renderCountries(mockCountries);
    expect(JSON.stringify(cleaned)).toEqual(JSON.stringify(expected));
  });
});