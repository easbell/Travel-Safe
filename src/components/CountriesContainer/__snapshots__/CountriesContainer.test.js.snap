// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountriesContainer should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CountriesContainer
    countries={
      Array [
        Object {
          "name": "USA",
        },
        Object {
          "name": "France",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <FilterResults
          defaultAllItems={true}
          fuseConfig={
            Object {
              "keys": Array [
                "name",
              ],
              "matchAllTokens": true,
              "threshold": 0,
              "tokenize": true,
            }
          }
          items={
            Array [
              Object {
                "name": "USA",
              },
              Object {
                "name": "France",
              },
            ]
          }
          prefilters={Array []}
        >
          [Function]
        </FilterResults>,
      ],
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": [Function],
          "defaultAllItems": true,
          "fuseConfig": Object {
            "keys": Array [
              "name",
            ],
            "matchAllTokens": true,
            "threshold": 0,
            "tokenize": true,
          },
          "items": Array [
            Object {
              "name": "USA",
            },
            Object {
              "name": "France",
            },
          ],
          "prefilters": Array [],
        },
        "ref": null,
        "rendered": [Function],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <FilterResults
            defaultAllItems={true}
            fuseConfig={
              Object {
                "keys": Array [
                  "name",
                ],
                "matchAllTokens": true,
                "threshold": 0,
                "tokenize": true,
              }
            }
            items={
              Array [
                Object {
                  "name": "USA",
                },
                Object {
                  "name": "France",
                },
              ]
            }
            prefilters={Array []}
          >
            [Function]
          </FilterResults>,
        ],
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": [Function],
            "defaultAllItems": true,
            "fuseConfig": Object {
              "keys": Array [
                "name",
              ],
              "matchAllTokens": true,
              "threshold": 0,
              "tokenize": true,
            },
            "items": Array [
              Object {
                "name": "USA",
              },
              Object {
                "name": "France",
              },
            ],
            "prefilters": Array [],
          },
          "ref": null,
          "rendered": [Function],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
